### PR TITLE
[MM-20489] Migrate failed_post_options and its tests to typescript

### PR DESCRIPTION
**Pull request by abdusabri**
#### Summary

- Migrates `failed_post_options` module and its tests to TypeScipt

- Adds `getPostMock` function to `utils/test_helper`

- Uses the newly added `getPostMock` function in `rhs_thread.test`

- Uses `getChannelMock` function in `rhs_thread.test`

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15452, https://mattermost.atlassian.net/browse/MM-20489

### CODE REVIEW

File: utils/test_helper.ts [hunk 1]
```diff
@@ -185,4 +186,34 @@ export class TestHelper {
...
+            message: 'post message',
+            parent_id: '',
+            props: {},
+            root_id: '',
+            type: 'system_add_remove',
+            update_at: 0,
+            user_id: 'user_id',
+        };
+        return Object.assign({}, defaultPost, override);
+    }
```
Comment (agarciamontoro) [inline utils/test_helper.ts:218]: This is a great addition, thank you!

**Review by agarciamontoro — APPROVED** [review]
This is amazing, @abdusabri! Thank you for your contribution! :tada:

**Review by crspeller — APPROVED** [review]

**Review by lindalumitchell — APPROVED** [review]

**Comment by abdusabri** [comment]
@devinbinnie or @agarciamontoro can you merge this PR so that other PRs can merge master and use the util function? 🙂

Thanks!

**Comment by agarciamontoro** [comment]
I'll merge this as soon as tests pass. Thank you again, @abdusabri :)

**Pull request merged**